recipes/redis-plus-plus/all/conanfile.py

```diff
@@ -0,0 +1,76 @@
+from conans import ConanFile, CMake, tools
+from conans.errors import ConanInvalidConfiguration
+import os
+
+
+class RedisPlusPlusConan(ConanFile):
+    name = "redis-plus-plus"
+    homepage = "https://github.com/sewenew/redis-plus-plus"
+    description = "Redis client written in C++"
+    topics = ("conan", "database", "redis", "client", "tls")
+    url = "https://github.com/conan-io/conan-center-index"
+    license = "Apache-2.0"
+    exports_sources = ["CMakeLists.txt"]
+    generators = "cmake", "cmake_find_package"
+    settings = "os", "compiler", "build_type", "arch"
+    options = {"shared": [True, False], "fPIC": [True, False], "with_tls": [True, False]}
+    default_options = {"shared": False, "fPIC": True, "with_tls": False}
+
+    _cmake = None
+
+    @property
+    def _source_subfolder(self):
+        return "source_subfolder"
+
+    @property
+    def _build_subfolder(self):
+        return "build_subfolder"
+
+    def config_options(self):
+        if self.settings.os == "Windows":
+            del self.options.fPIC
+
+    def configure(self):
+        if self.options.shared:
+            del self.options.fPIC
+
+        if self.settings.compiler == "Visual Studio" and self.options.shared:
+            raise ConanInvalidConfiguration("redis-plus-plus does not support begin compiled as shared library with Visual Studio")
```

Comment:
```suggestion

        if self.settings.compiler == "Visual Studio" and self.options.shared:
            raise ConanInvalidConfiguration("redis-plus-plus does not support begin compiled as shared library with Visual Studio")
```

Actually, hiredis-plus-plus support Visual Studio shared, but its CMakeLists.txt doesn't properly propagate dependencies of hiredis, leading to undefind symbols with shared Visual Studio. You can patch upstream CMakeLists to use hiredis config file instead of their fragile way to find and link hiredis.

Reply:
I saw this as well and I wanted to leave it for an upstream contribution on my part https://github.com/sewenew/redis-plus-plus/issues/153

Right now using the latest version it's painful to consume as shared on windows, if someone else needs this configuration they can do the working patching but it's not my focus for this PR =)

Reply:
openssl should be removed from the requirements in this recipe, because it's misleading, and therefore upstream CMakeLists should be patched. With this patch, shared on windows will be fixed for free. Please patch it in this PR, to no start with misleading dependencies or error messages.